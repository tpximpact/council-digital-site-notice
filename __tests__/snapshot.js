import { render } from "@testing-library/react";
import Home from "../src/app/(main)/page";

jest.mock("next/link", () => {
  return ({ href, children, className }) => {
    return (
      <a href={typeof href === "string" ? href : "#"} className={className}>
        {children}
      </a>
    );
  };
});

jest.mock("../src/app/actions/sanityClient", () => ({
  getActiveApplications: () => ({}),
  getActiveApplicationsByLocation: () => ({}),
  getGlobalContent: () => ({
    councilName: "Test Council",
    signUpUrl: "#",
  }),
}));

jest.mock("next/navigation", () => ({
  useSearchParams: () => new URLSearchParams({ page: "1" }),
  useRouter: () => jest.fn(),
  usePathname: () => jest.fn(),
}));

it("renders homepage unchanged", async () => {
  const { container } = render(await Home({ params: {}, searchParams: {} }));
  expect(container).toMatchSnapshot();
});
