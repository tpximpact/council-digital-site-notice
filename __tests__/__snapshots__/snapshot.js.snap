// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="govuk-main-wrapper"
    id="main"
  >
    <div
      class="govuk-width-container"
    >
      <h1
        class="govuk-heading-xl"
      >
        Find planning applications near you
      </h1>
      <p
        class="govuk-body-m"
      >
        Find, review and leave your comments on planning applications in
         
        Test Council
      </p>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-one-half"
        >
          <form
            action="/"
            method="get"
          >
            <div
              class="govuk-form-group"
            >
              <h2
                class="govuk-label-wrapper"
              >
                <label
                  class="govuk-label govuk-label--l"
                  for="postcode"
                >
                  Enter a postcode to find planning applications nearby
                </label>
              </h2>
              <input
                class="govuk-input"
                id="postcode"
                name="postcode"
                type="text"
                value=""
              />
            </div>
            <button
              class="govuk-button govuk-button--start "
              draggable="false"
              type="submit"
            >
              Search
              <svg
                aria-hidden="true"
                class="govuk-button__start-icon"
                focusable="false"
                height="19"
                viewBox="0 0 33 40"
                width="17.5"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h13l20 20-20 20H0l20-20z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </form>
        </div>
        <div
          class="govuk-grid-column-one-half"
        >
          <a
            class="govuk-button govuk-button--secondary"
            href="#"
          >
            Sign up for alerts on applications near you
          </a>
        </div>
      </div>
      <div
        class="govuk-grid-row"
        role="alert"
      >
        <div
          class="govuk-grid-column-full"
        >
          <h2
            class="govuk-heading-s"
          >
            No applications match your search
          </h2>
          <p
            class="govuk-body"
          >
            You can try searching again, or
             
            <a
              class="govuk-link govuk-link--no-visited-state"
              href="/"
            >
              go back
            </a>
          </p>
          <p
            class="govuk-body"
          >
            If you are having problems with finding what you need, you can:
          </p>
          <ul
            class="govuk-list govuk-list--bullet"
          >
            <li>
              Check you have spelled everything correctly
            </li>
            <li>
              Try being less specific with your search query
            </li>
          </ul>
        </div>
      </div>
    </div>
  </main>
</div>
`;
